librad/src/git/storage/pool.rs

```diff
@@ -63,41 +64,65 @@ impl From<Object<Storage, Error>> for PooledRef {
     }
 }
 
+#[derive(Clone)]
+pub struct Initialised(Arc<RwLock<bool>>);
+
+impl Initialised {
+    pub fn no() -> Self {
+        Self(Arc::new(RwLock::new(false)))
+    }
+}
+
 #[derive(Clone)]
 pub struct Config<S> {
     paths: Paths,
     signer: S,
+    init: Initialised,
```

Comment:
Can you elaborate a bit on what's going on with `Initialised`, please? From looking at the code, it only seems to track the `bool` value, but we call `mk_storage` regardless, so I'm a bit confused as to what magic it might be doing.

Reply:
It's sharing a mutex between different pool instances, to make sure only one storage instance initialises the repo.

Reply:
Ah, I see it now. So initially it will be false and one thread will have the exclusive lock initialising the repo via `Storage::with_fetchers`. After that `Storage::with_fetchers` will just open the repo since it exists and all threads will go through the first branch of the `if`.